src/compressor.js

```diff
@@ -0,0 +1,146 @@
+define(function (require) {
+	'use strict';
+
+	var p5sound = require('master');
+	var Effect = require('effect');
+  var CustomError = require('errorHandler');
+
+
+  /**
+   * Built on Web Audio Dynamics Compressor Node
+   * https://www.w3.org/TR/webaudio/#the-dynamicscompressornode-interface
+   *
+   * @class p5.Compressor
+   * @return {Object} Compressor Object
+   */
+	p5.Compressor = function() {
+		Effect.call(this);
+
+		this.compressor = this.ac.createDynamicsCompressor();
+    this.input.connect(this.compressor);
+    this.compressor.connect(this.wet);
+	};
+
+	p5.Compressor.prototype = Object.create(Effect.prototype);
+
+ 
+	p5.Compressor.prototype.process = function(src, attack, knee, 
+                                      ratio, threshold, release) {
+		src.connect(this.input);
+		this.set(attack, knee, ratio, threshold, release);
+	};
+
+  /**
+   * Set the paramters of the compressor
+   * @param {Number} attack    [default = .003, range 0 - 1]
+   * @param {Number} knee      [default = 30, range 0 - 40]
+   * @param {Number} ratio     [default = 12, range 1 - 20]
+   * @param {Number} threshold [default = -24, range -100 - 0]
+   * @param {Number} release   [default = .25, range 0 - 1]
+   */
+  p5.Compressor.prototype.set = function (attack, knee, 
+                                ratio, threshold, release) {
+
+    if (attack != 'undefined') {this.attack(attack);}
+    if (knee != 'undefined') {this.knee(knee);}
+    if (ratio != 'undefined') {this.ratio(ratio);}
+    if (threshold != 'undefined') {this.threshold(threshold);}
+    if (release != 'undefined') {this.release(release);}
+  };
+
+
+
+  /**
+   * Setter and Getter methdds 
+   * @method  {attack} set attack w/ time ramp or get current attack
+   * @method  {knee} set knee w/ time ramp or get current knee
+   * @method {ratio} set ratio w/ time ramp or get current ratio
+   * @method {threshold} set threshold w/ time ramp or get current threshold
+   * @method {release} set release w/ time ramp or get current release
+   */
+
+  p5.Compressor.prototype.attack = function (attack, time){
+    var self = this;
```

Comment:
The context isn't changing in these method, so we can just use `this` without creating the variable `self`. 

`var self = this` creates a reference to this instance of the compressor.

We would typically create a reference if we were referring to this compressor in a callback function or some other context.